src/js/media.js

```diff
@@ -1,41 +1,57 @@
 ( function( $ )  {
 
 	/**
-	 * 
+	 *
 	 * @param {Element} The button being clicked
 	 * @param {string} Which endpoint to query
 	 * @param {Function|bool} Optional callback to run after the request completes.
-	 *  
+	 *
 	 */
 	const handleClick = ( { button, endpoint, callback = false } ) => {
 		const postID = button.getAttribute( 'data-id' );
 		const [ spinner ] = button.parentNode.getElementsByClassName( 'spinner' );
+		const [ errorContainer ] = button.parentNode.getElementsByClassName( 'error' );
+		const path = `${ endpoint }${ postID }`;
 		const { __ } = wp.i18n;
-		
+		const { get } = window.lodash;
+
 		button.setAttribute( 'disabled', 'disabled' );
 		spinner.style.display = 'inline-block';
 		spinner.classList.add( 'is-active' );
 
-		const path = `${ endpoint }${ postID }`;
-		wp.apiRequest( { path } ).then( ( response ) => {
-			button.removeAttribute( 'disabled' );
-			spinner.style.display = 'none';
-			spinner.classList.remove( 'is-active' );
-			button.textContent = __( 'Rescan', 'classifai' );
-			callback && callback( response );
-		} );
+		wp.apiRequest( { path } )
+			.then(
+				( response ) => {
+					button.removeAttribute( 'disabled' );
+					spinner.style.display = 'none';
+					spinner.classList.remove( 'is-active' );
+					button.textContent = __( 'Rescan', 'classifai' );
+					callback && callback( response );
+				},
+				( error ) => {
+					const errorObj = get( error, 'responseJSON', {
+						code: 'unknown_error',
+						message: __( 'An unknown error occurred.', 'classifai' ),
+					} );
+					spinner.style.display = 'none';
+					spinner.classList.remove( 'is-active' );
+					button.textContent = __( 'Error', 'classifai' );
+					errorContainer.style.display = 'inline-block';
+					errorContainer.textContent = errorObj.message;
```

Comment:
How do users retry scanning after failing (due to network issue for example)?

Reply:
@dinhtungdu Thanks for the review! Just pushed a fix for this, so if an error happens, we make sure the button is no longer disabled, so it can be clicked again